macro/src/dialect/operation/builder.rs

```diff
@@ -1,82 +1,99 @@
+use std::iter::repeat;
+
 use super::{
     super::{error::Error, utility::sanitize_snake_case_name},
     FieldKind, Operation, OperationField,
 };
 use convert_case::{Case, Casing};
 use proc_macro2::{Ident, TokenStream};
 use quote::{format_ident, quote};
+use syn::GenericArgument;
 
 #[derive(Debug)]
 struct TypeStateItem {
     field_name: String,
-    yes: Ident,
-    no: Ident,
-    any: Ident,
+    generic_param: GenericArgument,
 }
 
 impl TypeStateItem {
-    pub fn new(class_name: &str, field_name: &str) -> Self {
-        let new_field_name = field_name.to_string().to_case(Case::Pascal);
+    pub fn new(field_name: String) -> Self {
         Self {
-            field_name: field_name.to_string(),
-            yes: format_ident!("{}__Yes__{}", class_name, new_field_name),
-            no: format_ident!("{}__No__{}", class_name, new_field_name),
-            any: format_ident!("{}__Any__{}", class_name, new_field_name),
+            generic_param: {
+                let ident = format_ident!("__{}", field_name.to_case(Case::Snake));
```

Comment:
I think it still makes sense to name those following the nominal convention of type parameters in Rust although users never touch them manually.

```suggestion
                let ident = format_ident!("{}", field_name.to_case(Case::Pascal));
```

I was also thinking naming those just `T1`, `T2`, ... if we use general `Set` and `Unset` types already because users won't actually distinguish those types by their names but only by their positions. What do you think?

Reply:
The problem with using ` format_ident!("{}", field_name.to_case(Case::Pascal));` is that this might clash with existing types. For example, if a field is called `option`, it will clash with `std::option::Option`.

`T1`, `T2`, ... could work though, I think that's a good idea.